### PR TITLE
[BTS-151] Modal Component

### PR DESCRIPTION
**Pull request by allyssonsantos**
modal

### CODE REVIEW

**Comment by rapahaeru** [comment]
The close button, should've a width and height of 24px.
And the color currently is #999 when in the zeplin is `rgba(128, 128, 128, 0.5)`.

Ps: I know that is a button component that was implemented. But i dont know if this patterns was discussed before. I only reporting the difference here.

What do you think?

Captura de tela de 2019-03-25 14-02-06

**Comment by rapahaeru** [comment]
Question: should `ESC` button close the modal? As well as clicking off from modal?

**Comment by allyssonsantos** [comment]
> The close button, should've a width and height of 24px.
> And the color currently is #999 when in the zeplin is `rgba(128, 128, 128, 0.5)`.
> 
> Ps: I know that is a button component that was implemented. But i dont know if this patterns was discussed before. I only reporting the difference here.
> 
> What do you think?
> 
> Captura de tela de 2019-03-25 14-02-06

Fixed! it was wrong.

**Comment by allyssonsantos** [comment]
> Question: should `ESC` button close the modal? As well as clicking off from modal?

@rapahaeru 

I think that two events needs to call the `onClose` prop function, not hiding the modal, the modal visibility is controlled by the user, not by the component itself.

What do you think?

File: components/Modal/Modal.unit.test.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    it('should call onClock when CloseIcon is clicked', () => {
+      const onCloseMock = jest.fn();
+      const modal = shallow(<Modal onClose={onCloseMock} />);
+
+      modal.find('CloseIcon').simulate('click');
+
+      expect(onCloseMock).toHaveBeenCalled();
+    });
+  });
```
[the remark this answers is not in the document]
Comment (allyssonsantos) [reply components/Modal/Modal.unit.test.jsx]: The cancel button isn't provided by default from the Modal Component, the user needs to implement it.

File: components/Modal/Modal.unit.test.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  describe('closeIcon Button', () => {
+    it('should exists a closeIcon button when modal is rendered', () => {
+      const modal = shallow(<Modal />);
+
+      expect(modal.find('CloseIcon')).toBeTruthy();
+    });
+
+    it('should call onClock when CloseIcon is clicked', () => {
```
[the remark this answers is not in the document]
Comment (allyssonsantos) [reply components/Modal/Modal.unit.test.jsx]: Done!

File: components/Modal/sub-components/Content.jsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import styled from 'styled-components';
+import Card from '../../Card';
+
+const Content = styled(Card.Content)`
+  padding: 16px 24px;
+  max-height: 70vh;
+  overflow-y: auto;
```
[the remark this answers is not in the document]
Comment (allyssonsantos) [reply components/Modal/sub-components/Content.jsx]: Fixed

File: components/Modal/Modal.jsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  constructor(props) {
+    super(props);
+
+    this.modalWrapper = document.createElement('section');
+    this.modalWrapper.style.cssText = `
+      align-items: center;
+      background-color: ${Colors.SHADOW[50]};
+      display: flex;
+      height: 100vh;
+      left: 0;
```
[the remark this answers is not in the document]
Comment (allyssonsantos) [reply components/Modal/Modal.jsx]: Removed

File: components/Modal/Modal.unit.test.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (rapahaeru) [reply components/Modal/Modal.unit.test.jsx]: Make sense.

**Pull request merged**